src/test/java/org/folio/dcb/controller/TransactionApiControllerTest.java

```diff
@@ -1498,4 +1523,179 @@ void createLendingCirculationRequestTestWithSpecialCharBarcode() throws Exceptio
       .andExpect(jsonPath("$.item").value(expectedDCBItem))
       .andExpect(jsonPath("$.patron").value(createDefaultDcbPatron()));
   }
+
+  @Test
+  void createBorrowingPickupSelfBorrowingPageRequestCirculationRequestTest() throws Exception {
+    removeExistedTransactionFromDbIfSoExists();
+    removeExistingTransactionsByItemId(ITEM_ID);
+
+    DcbItem expected = createDcbItem();
+    InventoryItem inventoryItem = createInventoryItem(ITEM_ID, HOLDING_RECORD_ID, "DCB_ITEM", ItemStatus.NameEnum.AVAILABLE);
+
+    Mockito.doReturn(ResultList.asSinglePage(inventoryItem)).when(inventoryItemStorageClient).fetchItemByQuery(anyString());
+    HoldingsStorageClient.Holding holding =
+            HoldingsStorageClient.Holding.builder().id(HOLDING_RECORD_ID).instanceId(INSTANCE_ID).build();
+    Mockito.doReturn(holding).when(holdingsStorageClient).findHolding(HOLDING_RECORD_ID);
+
+    this.mockMvc.perform(
+                    post("/transactions/" + DCB_TRANSACTION_ID)
+                            .content(asJsonString(createDcbTransactionByRoleAndSelfBorrowing(BORROWING_PICKUP,
+                                    Boolean.TRUE)))
+                            .headers(defaultHeaders())
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(status().isCreated())
+            .andExpect(jsonPath("$.status").value("CREATED"))
+            .andExpect(jsonPath("$.item").value(expected))
+            .andExpect(jsonPath("$.patron").value(createDcbPatronWithExactPatronId(EXISTED_PATRON_ID)));
```

Comment:
Can I ask you to format this file? These indentations are too large and do not comply with other tests here